modules/private.php

```diff
@@ -0,0 +1,298 @@
+	}
+
+	/**
+	 * Replaces the the blog's "name" value with "Protected Blog"
+	 *
+	 * @see privatize_blog_maybe_mask_blog_name()
+	 * @param mixed $value The requested non-URL site information.
+	 * @param mixed $what  Type of information requested.
+	 */
+	static function privatize_blog_mask_blog_name( $value, $what ) {
+		if ( in_array( $what, array( 'name', 'title' ), true ) ) {
+			$value = __( 'Protected Blog', 'jetpack' );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Filters new comments so that users can't comment on private blogs
+	 *
+	 * @param array $comment Documented in wp-includes/comment.php.
+	 */
+	static function privatize_blog_comments( $comment ) {
+		Jetpack_Private::privatize_blog( null );
+		return $comment;
+	}
+
+	/**
+	 * Extend the 'Site Visibility' privacy options to also include a private option
+	 */
+	static function privatize_blog_priv_selector() {
+		?>
+		<br /><input id="blog-private" type="radio" name="blog_public" value="-1" <?php checked( '-1', get_option( 'blog_public' ) ); ?> />
+		<label for="blog-private"><?php esc_html_e( 'I would like my site to be private, visible only to myself and users I choose', 'jetpack' ); ?></label>
+		<?php
+	}
+
+	/**
+	 * Don't let search engines index private sites
+	 * or sites not deemed publicly available, like deleted, archived, spam.
+	 *
+	 * @param string $output Robots.txt output.
+	 */
+	static function private_robots_txt( $output ) {
+		if ( Jetpack_Private::is_private_blog() ) {
+			$output  = "User-agent: *\n"; // Purposefully overriding current output; we only want these rules.
+			$output .= "Disallow: /\n";
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Filters the link title attribute for the message displayed in the 'At a Glance' dashboard widget.
+	 *
+	 * @param string $text Default attribute text.
+	 */
+	static function privatize_privacy_on_link_title( $text ) {
+		if ( Jetpack_Private::is_private_blog() ) {
+			return __( 'Your site is visible only to registered members', 'jetpack' );
+		}
+
+		return $text;
+	}
+
+	/**
+	 * Filters the link label for the message displayed in the 'At a Glance' dashboard widget.
+	 *
+	 * @param string $text Default text.
+	 */
+	static function privatize_privacy_on_link_text( $text ) {
+		if ( Jetpack_Private::is_private_blog() ) {
+			return __( 'Private', 'jetpack' );
+		}
+
+		return $text;
+	}
+
+	/**
+	 * Output the meta tag that tells Pinterest not to allow users to pin
+	 * content from this page.
+	 * https://support.pinterest.com/entries/21063792-what-if-i-don-t-want-images-from-my-site-to-be-pinned
+	 */
+	static function private_no_pinning() {
+		echo '<meta name="pinterest" content="nopin" />';
+	}
+
+	/**
+	 * Prevents ajax requests on private blogs for users who don't have permissions
+	 *
+	 * @param string    $action The Ajax nonce action.
+	 * @param false|int $result The result of the nonce check.
+	 */
+	static function private_blog_ajax_nonce_check( $action, $result ) {
```

Comment:
I ran into the following warning:

```
PHP Warning:  Missing argument 2 for Jetpack_Private::private_blog_ajax_nonce_check(), called in /srv/users/user6c5aabf7/apps/user6c5aabf7/public/wp-includes/class-wp-hook.php on line 288 and defined in wp-content/plugins/jetpack-dev/modules/private.php on line 230
```

Reply:
Whoops, yeah I forgot to pass the second parameter. This is fixed now. Good catch.